WordPress/Classes/ViewRelated/EEUUSCompliance/CompliancePopoverCoordinator.swift

```diff
@@ -1,52 +1,86 @@
 import UIKit
 
 protocol CompliancePopoverCoordinatorProtocol: AnyObject {
-    func presentIfNeeded(on viewController: UIViewController)
+    func presentIfNeeded()
     func navigateToSettings()
     func dismiss()
 }
 
 final class CompliancePopoverCoordinator: CompliancePopoverCoordinatorProtocol {
-    private weak var presentingViewController: UIViewController?
+
+    // MARK: - Dependencies
+
     private let complianceService = ComplianceLocationService()
     private let defaults: UserDefaults
 
+    // MARK: - Views
+
+    private static var window: UIWindow?
+
+    private let presentingViewController = UIViewController()
+
+    // MARK: - Init
+
     init(defaults: UserDefaults = UserDefaults.standard) {
         self.defaults = defaults
     }
 
-    func presentIfNeeded(on viewController: UIViewController) {
+    func presentIfNeeded() {
         guard FeatureFlag.compliancePopover.enabled, !defaults.didShowCompliancePopup else {
             return
         }
         complianceService.getIPCountryCode { [weak self] result in
-            if case .success(let countryCode) = result {
-                guard let self, self.shouldShowPrivacyBanner(countryCode: countryCode) else {
-                    return
-                }
-                DispatchQueue.main.async {
-                    self.presentPopover(on: viewController)
-                }
+            guard let self, case .success(let countryCode) = result, self.shouldShowPrivacyBanner(countryCode: countryCode) else {
+                return
+            }
+            DispatchQueue.main.async {
+                self.presentPopover()
             }
         }
     }
 
     func navigateToSettings() {
-        presentingViewController?.dismiss(animated: true) {
+        self.dismiss {
             RootViewCoordinator.sharedPresenter.navigateToPrivacySettings()
         }
     }
 
     func dismiss() {
-        presentingViewController?.dismiss(animated: true)
+        self.dismiss(completion: nil)
     }
 
+    // MARK: - Helpers
+
     private func shouldShowPrivacyBanner(countryCode: String) -> Bool {
         let isCountryInEU = Self.gdprCountryCodes.contains(countryCode)
         return isCountryInEU && !defaults.didShowCompliancePopup
     }
 
-    private func presentPopover(on viewController: UIViewController) {
+    private func dismiss(completion: (() -> Void)? = nil) {
+        self.presentingViewController.dismiss(animated: true) {
+            self.removeWindow()
```

Comment:
`completion?()` call is missing here, making "Go to Settings" button non-functional.